spec/models/concerns/payment_utility_spec.rb

```diff
@@ -0,0 +1,110 @@
+require 'rails_helper'
+
+
+# We use the User class since it includes the PaymentUtility class
+
+RSpec.describe User, type: :model do
+
+  let(:MembershipFee) { Payment::PAYMENT_TYPE_MEMBER }
+  let(:BrandingFee) { Payment::PAYMENT_TYPE_BRANDING }
+
+  let(:jul_1) { Time.zone.local(2018,  7, 1) }
+
+  let(:dec_1) { Time.zone.local(2018, 12, 1) }
+  let(:dec_2) { Time.zone.local(2018, 12, 2) }
+  let(:dec_3) { Time.zone.local(2018, 12, 3) }
+
+  let(:nov_30_next_year) { Time.zone.local(2019, 11, 30) }
+
+  let(:lastyear_dec_2)  { Time.zone.local(2017, 12, 2) }
+
+  let(:success) { Payment.order_to_payment_status('successful') }
+
+
+  around(:each) do |example|
+    Timecop.freeze(dec_1)
+    example.run
+    Timecop.return
+  end
+
+
+  describe '.next_payment_dates' do
+
+    context 'entity not found' do
+
+      describe 'returns empty array' do
+        context 'membership fee' do
+         pending # expect(described_class.next_payment_dates(100, Payment::PAYMENT_TYPE_MEMBER)).to eq []
+        end
+        context 'branding fee' do
+          pending # expect(described_class.next_payment_dates(100, Payment::PAYMENT_TYPE_BRANDING)).to eq []
+        end
+      end
+    end
+
+    context 'entity found' do
+
+      context 'no payments' do
+        let(:user) { create(:user) }
+        let(:company) { create(:company) }
+
+        describe 'membership fee' do
+         pending
+         # it 'start date = today, expire = 1 year + 1 day of start date' do
```

Comment:
I have been using the term "expire date" to == the last day of the latest payment period.  So, if the start of the period is Jan, and the last is Dec 31, then the expire date is Dec 31.  And, on January 1 of the next year, the payment has expired.  

It is bit tricky because on the payment _expire_date_, the payment has not het expired, but will do so at midnight.

Reply:
This all points to the fact that we're at the point where we need to nail down definitions.  (And then refactor all of this stuff - PaymentUtility and the methods in User.

I'd like to leave this as is and then revise it once we've had the discussions and come up with definitions.